product_variant_default_code/models/product.py

```diff
@@ -100,12 +76,62 @@ class ProductTemplate(models.Model):
         "attribute name",
     )
 
+    variant_default_code_error = fields.Text(
+        compute="_compute_variant_default_code_error"
+    )
+
+    def is_automask(self):
+        return bool(
+            not self.user_has_groups(
+                "product_variant_default_code.group_product_default_code_manual_mask"
+            )
+        )
+
+    @api.depends(
+        "code_prefix",
+        "attribute_line_ids.value_ids",
+        "attribute_line_ids.value_ids.code",
+        "attribute_line_ids.value_ids.name",
+    )
+    def _compute_variant_default_code_error(self):
+        if self.is_automask():
+            for rec in self:
+                error_txt = ""
+                if not rec.code_prefix and self.is_automask():
```

Comment:
I think my explanation was not clear, I just say that you should init a variable outside the loop to avoid to call it all the time.
error on missing code should be always display even in case of automask

```suggestion
        automask = self.is_automask():
        for rec in self:
            error_txt = ""
            if not rec.code_prefix and automask:
```

Reply:
My bad, i did not check all the 'is_automask()' uses